### PR TITLE
Feature/configure download url

### PR DESCRIPTION
**Pull request by dewthefifth**
Change will allow the download URL to be configured in the playbook for purposes of using local mirrors or proxies.

### CODE REVIEW

**Comment by ecristea** [comment]
Hi,
Thank you for the pull request.
Unfortunatelly the test fails and I cannot do the merge.
The test is far from ideal and I've been looking for a different solution (to test the role).
One issue with current test is that it's not clear why it fails.
Also I am interested to see if it's possible to add a test for the new feature.

**Comment by tecris** [comment]
Hi,
Thank you for the pull request.
Unfortunatelly the test fails and I cannot do the merge.
The test is far from ideal and I've been looking for a different solution (to test the role).
One issue with current test is that it's not clear why it fails.
Also I am interested to see if it's possible to add a test for the new feature.

**Comment by dewthefifth** [comment]
I'm not sure why it failed myself, as I did test the changes against a brand new CentOS 7 box. However, it's possible since the tests run against ubuntu I've messed something up. I'll run them against a new box again right now, just to make sure I haven't missed something obvious.

Do you have a quick primer I might be able to use to determine how to run the tests?

For reference sake I'm running
ansible 2.3.0.0
  config file = /cygdrive/c/Projects/rigors/Source/ansible-scripts/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Mar 13 2017, 20:56:15) [GCC 5.4.0]

**Comment by dewthefifth** [comment]
TLDR; All checks have passed.

The tests work by counting the number of items which have changed and the number of items that have failed. No failures were occurring, but because I split unarchive into get_url and unarchive the number of changes increased from 2 to 3.

That said, when I started looking into the failures I realized that the tests were only counting the number of changes during the idempotence run, which should not have changed since both 'get_url' and 'unarchive' should both be skipped if maven was already installed. I realized that the 'get_url' was executing instead of changing, and included a commit to force a skip if maven is already installed.
